ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ip_access_list.py

```diff
@@ -0,0 +1,184 @@
+                ],
+            }
+            return outbound_access_list
+        return None
+
+    def _get_access_lists_inbound(self) -> List | None:
+        """
+        Returns the inbound access lists for different interfaces.
+        The ACL name per interface is of the form "WAN_TRANSPORT_<INTERFACE_NAME>_IN".
+        """
+        inbound_access_lists = []
+        for l3_interface in self.shared_utils.wan_interfaces:
+            interface_ip = l3_interface.get("ip_address")
+            # Need to see how to handle dhcp cases.
+            if interface_ip and interface_ip != "dhcp":
+                subnet = str(ip_network(interface_ip, False))
+                l3_interface_name = l3_interface.get("name")
+                inbound_access_list = {
+                    "name": self._get_inbound_wan_acl_name(l3_interface_name),
+                    "entries": [],
+                }
+                # Allow BGP
+                inbound_access_list["entries"].extend(
+                    [
+                        {
+                            "sequence": self._get_sequence(refresh=True),
+                            "remark": "### Allow BGP ###",
+                        },
+                        {
+                            "sequence": self._get_sequence(),
+                            "action": "permit",
+                            "protocol": "tcp",
+                            "source": "any",
+                            "destination": subnet,
+                            "destination_ports_match": "eq",
+                            "destination_ports": ["bgp"],
+                        },
+                    ]
+                )
+
+                # Allow GRE
+                inbound_access_list["entries"].extend(
+                    [
+                        {
+                            "sequence": self._get_sequence(),
+                            "remark": "### Allow GRE ###",
+                        },
+                        {
+                            "sequence": self._get_sequence(),
+                            "action": "permit",
+                            "protocol": "gre",
+                            "source": "any",
+                            "destination": subnet,
+                        },
+                    ]
+                )
+
+                # Allow IPSEC
+                inbound_access_list["entries"].extend(
+                    [
+                        {
+                            "sequence": self._get_sequence(),
+                            "remark": "### Allow IPSEC ###",
+                        },
+                        {
+                            "sequence": self._get_sequence(),
+                            "action": "permit",
+                            "protocol": "udp",
+                            "source": "any",
+                            "destination": subnet,
```

Comment:
IMO we should permit only to the specific interface IP address (for all rules here)